### PR TITLE
[Feature](array-function) Support array functions for nested type datev2 and datetimev2

### LINKED ISSUE #15952: [Feature] Support array functions for nested type datev2/datetimev2
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/doris/issues?q=is%3Aissue) and found no similar issues.

### Description

_No response_

### Use case

_No response_

### Related issues

_No response_

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by Yukang-Lian**
# Proposed changes

Issue Number: close #15952 

## Problem summary

Describe your changes.

## Checklist(Required)

1. Does it affect the original behavior: 
    - [ ] Yes
    - [ ] No
    - [ ] I don't know
2. Has unit tests been added:
    - [ ] Yes
    - [ ] No
    - [ ] No Need
3. Has document been added or modified:
    - [ ] Yes
    - [ ] No
    - [ ] No Need
4. Does it need to update dependencies:
    - [ ] Yes
    - [ ] No
5. Are there any changes that cannot be rolled back:
    - [ ] Yes (If Yes, please explain WHY)
    - [ ] No

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by hello-stephen** [comment]
TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 34.12 seconds
 load time: 496 seconds
 storage size: 17122484656 Bytes
 https://doris-community-test-1308700295.cos.ap-hongkong.myqcloud.com/tmp/20230202194406_clickbench_pr_89813.html

File: be/src/vec/functions/array/function_array_remove.h [hunk 1]
```diff
@@ -322,13 +330,21 @@ class FunctionArrayRemove : public IFunction {
                 res = _execute_number_expanded<ColumnDecimal128>(offsets, *nested_column,
                                                                  *right_column, nested_null_map);
             }
-        } else if (is_date_or_datetime(right_type) && is_date_or_datetime(left_element_type)) {
+        } else if ((is_date_or_datetime(right_type) && is_date_or_datetime(left_element_type)) ||
+                   (is_date_v2_or_datetime_v2(right_type) &&
+                    is_date_v2_or_datetime_v2(left_element_type))) {
```
Comment (Gabriel39) [inline be/src/vec/functions/array/function_array_remove.h:335]: Why not use `_execute_number_expanded<ColumnDateV2>` if `is_date_v2_or_datetime_v2(right_type)` ?

File: fe/fe-common/src/main/java/org/apache/doris/common/Config.java [hunk 2]
```diff
@@ -1717,7 +1717,7 @@ public class Config extends ConfigBase {
      * If set to TRUE, FE will convert date/datetime to datev2/datetimev2(0) automatically.
      */
     @ConfField(mutable = true, masterOnly = true)
-    public static boolean enable_date_conversion = false;
+    public static boolean enable_date_conversion = true;
```
Comment (carlvinhust2012) [inline fe/fe-common/src/main/java/org/apache/doris/common/Config.java:1720]: Will the datev2/datetimev2 be enabled by default?

Comment (Yukang-Lian) [reply fe/fe-common/src/main/java/org/apache/doris/common/Config.java:1720]: no, this is just for test, will remove it soon.

**Review by Gabriel39 — APPROVED** [review]
LGTM

**Pull request merged**